### PR TITLE
Setup linting and formatting dev tools

### PR DESCRIPTION
**Pull request by fa9r**
I setup dev tools for linting and formatting to improve code quality of our .py files.

Changes:
- Added `pyproject.toml` that defines all dev tools and their configs. This also enables `poetry install` to set them up.
- Added scripts for linting, formatting, and spell-checking in `scripts/`.
- Defined pre-commit hooks in `.pre-commit-config.yaml` that calls linting and spell-checking.
- Moved all the .py code files into src/ and fixed all linting/formatting issues.

### CODE REVIEW

**Review by ayush714 — APPROVED** [review]
Really Nice work, It will be also good if you look at adding linting formatting tools in zenfiles as well, nevertheless, let's get this work for zenbytes, then we can take a look later on because it's not urgent.

File: 02 - Transitioning To Production.ipynb [hunk 1]
```diff
@@ -433,7 +431,7 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "!zenml served-models delete <UUID-of-model-server>\n",
+    "!zenml served-models delete 01f049c8-c713-4e35-96c4-9f3738440762\n",
```
Comment (htahir1) [inline 02 - Transitioning To Production.ipynb:434]: Maybe lets not hard-code the UUID

**Review by htahir1 — CHANGES_REQUESTED** [review]
One small comment but other than that looks good!

File: 02 - Transitioning To Production.ipynb (hunk 1 above)
Comment (fa9r) [reply 02 - Transitioning To Production.ipynb]: Good point. Fixed.

**Review by htahir1 — APPROVED** [review]

**Comment by htahir1** [comment]
Really awesome job!

**Pull request merged**